string.c

```diff
@@ -9257,6 +9257,80 @@ rb_str_delete_prefix(VALUE str, VALUE prefix)
     return rb_str_subseq(str, prefixlen, RSTRING_LEN(str) - prefixlen);
 }
 
+static long
+deleted_suffix_length(VALUE str, VALUE suffix)
+{
+    char *strptr, *suffixptr;
+    long olen, suffixlen;
+
+    StringValue(suffix);
+    if (is_broken_string(suffix)) return 0;
+    rb_enc_check(str, suffix);
+
+    /* return 0 if not start with suffix */
+    suffixlen = RSTRING_LEN(suffix);
+    if (suffixlen <= 0) return 0;
+    olen = RSTRING_LEN(str);
+    if (olen < suffixlen) return 0;
+    strptr = RSTRING_PTR(str);
+    suffixptr = RSTRING_PTR(suffix);
+    if (memcmp(strptr + olen - suffixlen, suffixptr, suffixlen) != 0) return 0;
```

Comment:
I pushed, but it looks github is currently unstable. Please wait it will be reflected.

![image](https://user-images.githubusercontent.com/2290461/27552698-4ecbb2f2-5ae3-11e7-8bec-82fbbfdb9b99.png)